File: backend/admin-center/src/main/java/com/admin/entity/AdminCommonTable.java
package com.admin.entity;

import jakarta.persistence.*;
import lombok.*;

import java.time.Instant;

/**
 * Admin Center view of dw_common_table_definitions
 */
@Entity
@Table(name = "dw_common_table_definitions")
@Data
@NoArgsConstructor
@AllArgsConstructor
@EqualsAndHashCode(of = "id")
public class AdminCommonTable {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(name = "code", length = 100)
    private String code;

    @Column(name = "name", length = 200)
    private String name;

    @Column(name = "description", columnDefinition = "TEXT")
    private String description;

    @Column(name = "status", length = 20)
    private String status;

    @Column(name = "version", length = 20)
    private String version;

    @Column(name = "enabled")
    private Boolean enabled;

    @Column(name = "deployed_at")
    private Instant deployedAt;

    @Column(name = "deployed_by", length = 100)
    private String deployedBy;

    @Column(name = "updated_by", length = 100)
    private String updatedBy;

    @Column(name = "created_by", length = 50)
    private String createdBy;

    @Column(name = "created_at")
    private Instant createdAt;

    @Column(name = "updated_at")
    private Instant updatedAt;
}

File: frontend/admin-center/src/api/adminCommonTable.ts
import request from './request'

export interface AdminCommonTable {
  id: number
  code: string
  name: string
  description?: string
  status: string
  version?: string
  enabled: boolean
  deployedAt?: string
  deployedBy?: string
  updatedBy?: string
  createdBy?: string
  createdAt?: string
  updatedAt?: string
}

export interface AdminCommonTableDeployment {
  id: number
  commonTableId: number
  version?: string
  status: string
  deployedAt?: string
  deployedBy?: string
  notes?: string
  createdAt?: string
}

export interface AdminCommonTableAccess {
  id: number
  commonTableId: number
  accessType: string
  targetType: string
  targetId: string
  roleName?: string
  createdAt?: string
  createdBy?: string
}

const BASE = '/admin/common-tables'

export const adminCommonTableApi = {
  list(): Promise<{ success: boolean; data: AdminCommonTable[] }> {
    return request.get(BASE)
  },

  getById(id: number): Promise<{ success: boolean; data: AdminCommonTable }> {
    return request.get(`${BASE}/${id}`)
  },

  setEnabled(id: number, enabled: boolean): Promise<{ success: boolean; data: AdminCommonTable }> {
    return request.put(`${BASE}/${id}/enabled`, { enabled })
  },

  delete(id: number): Promise<{ success: boolean }> {
    return request.delete(`${BASE}/${id}`)
  },

  listAllDeployments(): Promise<{ success: boolean; data: AdminCommonTableDeployment[] }> {
    return request.get(`${BASE}/deployments`)
  },

  listDeployments(id: number): Promise<{ success: boolean; data: AdminCommonTableDeployment[] }> {
    return request.get(`${BASE}/${id}/deployments`)
  },

  rollback(deploymentId: number, notes?: string): Promise<{ success: boolean; data: AdminCommonTableDeployment }> {
    return request.post(`${BASE}/deployments/${deploymentId}/rollback`, { notes })
  },

  getVersionsByCode(code: string): Promise<{ success: boolean; data: AdminCommonTable[] }> {
    return request.get(`${BASE}/code/${code}/versions`)
  },

  getAccess(id: number): Promise<{ success: boolean; data: AdminCommonTableAccess[] }> {
    return request.get(`${BASE}/${id}/access`)
  },

  addAccess(id: number, roleId: string): Promise<{ success: boolean; data: AdminCommonTableAccess }> {
    return request.post(`${BASE}/${id}/access`, { roleId })
  },

  deleteAccess(id: number, accessId: number): Promise<{ success: boolean }> {
    return request.delete(`${BASE}/${id}/access/${accessId}`)
  }
}
